src/app/view/components/dot-large-message-display/services/dot-large-message-display.service.ts

```diff
@@ -0,0 +1,64 @@
+import { BehaviorSubject, Observable } from 'rxjs';
+import { Injectable } from '@angular/core';
+import { DotcmsEventsService, DotEventData } from 'dotcms-js';
+
+export interface DotLargeMessageDisplayParams {
+    title: string;
+    width?: string;
+    height?: string;
+    body?: string;
+    code?: {
+        lang: string;
+        content: string;
+    };
+}
+
+@Injectable({
+    providedIn: 'root'
+})
+export class DotLargeMessageDisplayService {
+    private _messages: BehaviorSubject<DotLargeMessageDisplayParams> = new BehaviorSubject(null);
+
+    constructor(dotcmsEventsService: DotcmsEventsService) {
+        dotcmsEventsService.subscribeTo('LARGE_MESSAGE').subscribe((messageEvent: DotEventData) => {
+            const { code, width, body, title, lang, height } = messageEvent.data;
+            this.push({
+                title,
+                height,
+                width,
+                body,
+                code: { lang, content: code }
```

Comment:
Why `lang` is coming outside code?

Reply:
interfase structure:
![image](https://user-images.githubusercontent.com/37185433/54626790-89f98a80-4a37-11e9-9456-c280b6c09126.png)